tests/drivers/rtc/rtc_api/src/test_y2k.c

```diff
@@ -40,7 +40,15 @@ ZTEST(rtc_api, test_y2k)
 
 	/* Party like it's 1999 */
 	zassert_not_null(gmtime_r(&t[Y99], tm[Y99]));
-	zassert_ok(rtc_set_time(rtc, &rtm[Y99]));
+
+	int ret = rtc_set_time(rtc, &rtm[Y99]);
+
+	if (ret == -ENOTSUP) {
```

Comment:
The RTC should return -EINVAL if the time exceeds hardware capabilities :)

https://github.com/zephyrproject-rtos/zephyr/blob/37e3449a393710df6339bd97cfc73679c63546ca/include/zephyr/drivers/rtc.h#L195

Reply:
It should be fixed now.